includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Newspack Data Events Ga4 Connector
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Data_Events\Connectors;
+
+require_once 'class-event.php';
+
+use Automattic\Jetpack\Device_Detection;
+use Newspack\Data_Events;
+use Newspack\Data_Events\Connectors\GA4\Event;
+use Newspack\Logger;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main Class.
+ */
+class GA4 {
+
+	/**
+	 * Initialize the class and registers the handlers
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		Data_Events::register_handler( [ __CLASS__, 'handle_reader_logged_in' ], 'reader_logged_in' );
+	}
+
+	/**
+	 * Handler for the reader_logged_in event.
+	 *
+	 * @param int   $timestamp Timestamp of the event.
+	 * @param array $data      Data associated with the event.
+	 * @param int   $client_id ID of the client that triggered the event.
+	 *
+	 * @throws \Exception If the event is invalid.
+	 * @return void
+	 */
+	public static function handle_reader_logged_in( $timestamp, $data, $client_id ) {
+
+		$params = array_merge(
+			self::get_default_params(),
+			[ 'user_id' => $data['user_id'] ]
+		);
+
+		$event = new Event( 'reader_login', $params );
+		self::send_event( $event, $client_id );
+	}
+
+	/**
+	 * Get the default parameters that are added to all events
+	 *
+	 * @return array
+	 */
+	private static function get_default_params() {
+		$params = [
+			'logged_in' => is_user_logged_in() ? 'yes' : 'no',
```

Comment:
The handler executes outside of the user's original request. `is_user_logged_in()` or any other session related information is not to be trusted unless explicitly passed as raw data.

If the handler/connection requires additional information, the action payload should be revised, and make that data available for every handler. I don't think it's the case here because "logged in" is the actual information being sent.

Reply:
These can get quite extensive, I'm not sure we should attach too much session information unrelated to the actual action/event. As I commented below, doesn't GA have the ability to integrate the session information through the `_ga` cookie and allow you to contextualize the event?

Reply:
I'm looking into it. Currently it's not, but I'll dig a bit more.

But this will never include WP specific information, such as logged_in, is_subscribed, is_donor and other information

Reply:
True, for WP-specific data we can think of a generic and filterable `user_metadata` sent on every event. WDYT?

Reply:
Will work on this default data on the next PR